project/Build.scala

```diff
@@ -699,6 +699,42 @@ object Build {
       MyScalaJSPlugin
   ).settings(
       commonLinkerInterfaceSettings,
+
+      /* Add the sources of scalajs-logging to managed sources. This is outside
+       * of `target/` so that `clean` does not remove them, making IDE happier.
+       */
+      managedSourceDirectories in Compile +=
+        baseDirectory.value / "scalajs-logging-src",
+
+      // Source generator to retrieve the sources of scalajs-logging
+      sourceGenerators in Compile += Def.task {
+        val s = streams.value
+        val log = s.log
+
+        // Retrieve the source jar of scalajs-logging
+        val retrieveDir = baseDirectory.value / "scalajs-logging-src-jars"
+        val binVer = scalaBinaryVersion.value
+        val lm = dependencyResolution.value
+        val jars = lm.retrieve(
+            "org.scala-js" % s"scalajs-logging_$binVer" % "1.1.1" classifier "sources" intransitive(),
+            scalaModuleInfo = None, retrieveDir, log)
+          .fold(w => throw w.resolveException, _.distinct)
+        assert(jars.size == 1, jars.toString())
+        val jar = jars.head
+
+        // Extract it
+        val targetDir = baseDirectory.value / "scalajs-logging-src"
+        val cacheDir = s.cacheDirectory / "scalajs-logging-src-cache"
+        val fileSet = FileFunction.cached(cacheDir, FilesInfo.lastModified, FilesInfo.exists) { _ =>
+          s.log.info(s"Unpacking scalajs-logging sources to $targetDir...")
+          if (targetDir.exists)
+            IO.delete(targetDir)
+          IO.createDirectory(targetDir)
+          IO.unzip(jar, targetDir)
+        } (Set(jar))
+
+        fileSet.toSeq.filter(_.getPath().endsWith(".scala"))
+      }.taskValue,
```

Comment:
Is there a reason this is not the same code than the one to fetch scala sources?

Reply:
* Using `dependencyResolution` directly makes sure that the dependency does not leak in the POM files in any way (putting anything in `libraryDependencies` leaks into the POM). This is not a problem for scalalib because we don't publish that artifact; we only take its .sjsir files and stuff them in another artifact with its own POM.
* It allows the IDEs to understand what's going on. With the way scalalib does it, they wouldn't (but we don't need to navigate in the scalalib so it's fine there).

Reply:
I see. Could we use this approach for the scalalib as well instead? (I understand there are some considerations for the community build there).

Reply:
We could probably use that for the scalalib as well. We're not constrained by the community build anymore, since this repo is not part of it anymore. We have our own nightly-against-nightly at https://github.com/scala-js/scala-js-test-scala-nightly. As long as that works, we're good.